main.go

```diff
@@ -128,6 +139,13 @@ type logsConfig struct {
 	enabled bool
 }
 
+type tracesConfig struct {
+	writeEndpoint string
+	tenantHeader  string
+	// enable traces if writeEndpoint is provided.
+	enabled bool
```

Comment:
I added a TODO.  There is an argument that if we support neither metrics nor logs we shouldn't listen on HTTP at all, and I want to avoid the changes needed for that with this initial PR.

Reply:
Cool, makes sense :+1: